### PR TITLE
[Tests] Add a test for accessing globally-isolated Sendable mutable storage of a Sendable value type from outside of the module.

### PR DESCRIPTION
**Pull request by simanerush**
Resolves rdar://136761737.

### CODE REVIEW

**Comment by simanerush** [comment]
@swift-ci please smoke test

File: test/Concurrency/nonisolated_access.swift [hunk 1]
```diff
@@ -3,17 +3,18 @@
...
 
 // Build the client
 // RUN: %target-swift-frontend -emit-module %t/src/Client.swift \
-// RUN: -disable-availability-checking \
+// RUN: -disable-availability-checking -verify \
 // RUN: -module-name Client -I %t -swift-version 6 \
 // RUN: -emit-module-path %t/Client.swiftmodule
 
 // REQUIRES: concurrency
+// REQUIRES: asserts
```
Comment (AnthonyLatsis) [inline test/Concurrency/nonisolated_access.swift:17]: Why do we need asserts?

File: test/Concurrency/nonisolated_access.swift [hunk 2]
```diff
@@ -39,6 +41,8 @@ actor A {
   func test() {
     var s = S()
     s.x += 0 // okay
+    // expected-error@+1 {{main actor-isolated property 'y' can not be mutated on a nonisolated actor instance}}
```
Comment (AnthonyLatsis) [inline test/Concurrency/nonisolated_access.swift:44]: Just bringing to notice that this error is confusing — the instance is a struct, not an actor.

Comment (simanerush) [reply test/Concurrency/nonisolated_access.swift:44]: Yea, good point. I read this as actor instance is referring to actor A, but I can see your point upon a closer look

**Comment by simanerush** [comment]
@swift-ci please smoke test

File: test/Concurrency/nonisolated_access.swift (hunk 2 above)
Comment (AnthonyLatsis) [reply test/Concurrency/nonisolated_access.swift:43]: #76909.

**Pull request merged**